### PR TITLE
Update ScalarDB and other dependencies

### PR DESCRIPTION
**Pull request by brfrn169**
This PR update the ScalarDB and other dependencies in the samples. Please take a look!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,4 +7,6 @@ This repository contains sample applications for [ScalarDB](https://github.com/s
 - [Multi-storage Transaction Sample](multi-storage-transaction-sample/)
 - [Microservice Transaction Sample](microservice-transaction-sample/)
 - [ScalarDB GraphQL Sample](scalardb-graphql-sample/)
+- [Spring Data JDBC for ScalarDB Sample](spring-data-sample/)
+- [Spring Data JDBC for ScalarDB with Multi-storage Transactions Sample](spring-data-multi-storage-transaction-sample/)
```
Comment (brfrn169) [inline README.md:11]: Found the index didn't have the `Spring Data JDBC` samples.

File: microservice-transaction-sample/build.gradle [hunk 2]
```diff
@@ -3,12 +3,12 @@ subprojects {
     project.version = '1.0'
 
     ext {
-        grpcVersion = '1.38.0'
-        protocVersion = '3.17.2'
-        scalarDbVersion = '3.6.0'
-        picoCliVersion = '4.1.4'
+        grpcVersion = '1.53.0'
+        protocVersion = '3.23.1'
+        scalarDbVersion = '3.9.0'
```
Comment (brfrn169) [inline microservice-transaction-sample/build.gradle:8]: Use ScalarDB `3.9.0`.

File: spring-data-multi-storage-transaction-sample/README.md
Comment (brfrn169) [inline spring-data-multi-storage-transaction-sample/README.md:1]: @komamitsu Can you please check the `Spring Data JDBC for ScalarDB` part?

File: README.md (hunk 1 above)
Comment (komamitsu) [reply README.md:11]: Thanks! 🙇

File: spring-data-multi-storage-transaction-sample/src/main/java/sample/domain/repository/CustomerRepository.java [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
-import org.springframework.data.repository.PagingAndSortingRepository;
 import org.springframework.stereotype.Repository;
 import org.springframework.transaction.annotation.Transactional;
 import sample.domain.model.Customer;
 
 @Transactional
 @Repository
-public interface CustomerRepository
-    extends PagingAndSortingRepository<Customer, Integer>, ScalarDbHelperRepository<Customer> {
+public interface CustomerRepository extends ScalarDbRepository<Customer, Integer> {
```
Comment (komamitsu) [inline spring-data-multi-storage-transaction-sample/src/main/java/sample/domain/repository/CustomerRepository.java:11]: 👍

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by komamitsu — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**